packages/lesswrong/lib/collections/comments/views.ts

```diff
@@ -39,13 +44,11 @@ declare global {
    * In past versions, different subsets of these depending on whether you were
    * using an answers view, a subforum view, or something else.
    */
-  type CommentSortingMode =
-     "top"
-    |"groupByPost"
-    |"new"|"newest"
-    |"old"|"oldest"
-    |"magic"
-    |"recentComments"|"recentDiscussion"
+  export type CommentSortingMode = UnionOf<typeof COMMENT_SORTING_MODES>;
+}
+
+export const isCommentSortingMode = (value: string): value is CommentSortingMode => {
```

Comment:
I know the EA forum does this in some places but I'm not huge a fan of writing basically redundant wrapper functions around the TupleSet type guards (since the whole point of me creating TupleSets was to automatically turn set-inclusion checks into typescript type guards).  I'd probably just directly use `COMMENT_SORTING_MODES.has` in the relevant place in the code.

Reply:
okay, we worked through this and now there's no redundant wrapper function and the code for checking the comment sort by query is neatened up